Change ashwalkers to have custom lungs

coiax opened the pull request:
Ashwalkers now have lungs that are specially adapted to Lavaland
environments. However, that specialisation comes at a cost at sometimes
not being able to operate in normal human air conditions.

- Ashwalker lungs take a gas sample of the default lavaland atmosphere
  on initialisation in order to customise themselves
- Currently, excess O2/N2 has no functional drawback, so no maximum
  value is set on the lungs. If Lavaland rolls a higher N2/O2 mole count
  than human air, they will still struggle however.
- Ashwalkers are no longer NOBREATH, bringing them closer
  to regular lizards.

:cl: coiax
balance: Ashwalkers now have lungs adapted for the atmosphere of Indecipheres,
and now breathe. However, if the Indecipheres
atmosphere has miasma, they are immune to it.
/:cl:

coiax committed:
Changes ashwalkers to have custom lungs

Ashwalkers now have lungs that are specially adapted to Lavaland
environments. However, that specialisation comes at a cost at sometimes
not being able to operate in normal human air conditions.

- Ashwalker lungs take a gas sample of the default lavaland atmosphere
  on initialisation in order to customise themselves
- Currently, excess O2/N2 has no functional drawback, so no maximum
  value is set on the lungs. If Lavaland rolls a higher N2/O2 mole count
  than human air, they will still struggle however.
- Ashwalkers are no longer NOBREATH or VIRUSIMMUNE, bringing them closer
  to regular lizards.

dragomagol commented:
Shouldn't this have a changelog? Seems like something important for ghosts playing ashwalkers to know.

rasepretep commented:
Would this make it possible to get an Ashwalker (not ghost-spawned) corpse from a legion, epi it and get those lungs transplanted to gain that benefit? Alternatively, could this open a path to OOC character customization, allowing lizard players to not only pick regular/digitigrade legs, but regular/ashwalker lungs as well?

coiax commented:
> 
> 
> Shouldn't this have a changelog? Seems like something important for ghosts playing ashwalkers to know.

You're completely right.

D4C-420 commented:
Don't they have VIRUSIMMUNE because they don't have any reasonable way to clean blood and gibs and as a result will get sick from walking around in it? I mean it should be miners and fauna killing ash walkers not random illness they can't cure.

coiax commented:
As far as I understand, the sickness from gibs and blood comes from miasma buildup, which won't happen in the lavaland atmosphere, because it's constantly being reset to the normal air values.

And if miasma is normally part of the air, then they're immune to it.

MrMelbert commented:
as of #57957 , gibs / dead people no longer spit out miasma and instead try to infect you with diseases directly. Ashwalkers were given virus immunity a month or so ago because their tendril is often coated in gibs and dead bodies, which very often infected them with nasty viruses.

coiax commented:
> 
> 
> as of #57957 , gibs / dead people no longer spit out miasma and instead try to infect you with diseases directly. Ashwalkers were given virus immunity a month or so ago because their tendril is often coated in gibs and dead bodies, which very often infected them with nasty viruses they had no means of curing or preventing.

Did not know this, I will restore their virus immunity.

> Side question: If you take ash-walker mutation toxin on the station, will your lungs be functionally the same as normal lungs?

The lungs are "prototyped" on the ideal lavaland gas that the air controller stores, ie. it's not on "first breath" or anything. Ashwalkers (or more specifically, blackened frilled lungs) created anywhere will be based on lavaland atmosphere.
Hilariously, even if the map is icebox.

MrMelbert commented:
> The lungs are "prototyped" on the ideal lavaland gas 

Oh well, seems like gone are the days of sipping ashwalker mutation toxin in botany for nobreath and virus immunity. Still a cool change, though.

ATH1909 commented:
time to turn people into ash walkers as a changeling to make them suffocate 😎

ATH1909 commented on code/modules/surgery/organs/lungs.dm:
```diff
@@ -562,3 +564,54 @@
+	var/TOLERANCE = 5
+	safe_oxygen_min = max(0, oxygen_pp - TOLERANCE)
+	safe_nitro_min = max(0, nitrogen_pp - TOLERANCE)
+	safe_toxins_min = max(0, plasma_pp - TOLERANCE)
```
👀

coiax replied:
You're going to have to be more specific.

Mothblocks replied:
he never is

Mothblocks replied:
this is your only warning from me--stop making reviews like this

Mothblocks commented on code/modules/surgery/organs/lungs.dm:
```diff
@@ -562,3 +564,54 @@
+
+	// Normal oxygen is 21 kPa partial pressure, but SS13 humans can tolerate down
+	// to 16 kPa. So it follows that ashwalkers, as humanoids, follow the same rules.
+	var/TOLERANCE = 5
```
Can you make this a define? I don't like SCREAMING_VARIABLES when they're just normal variables, confusing.

Mothblocks commented on code/modules/mob/living/carbon/human/species_types/lizardpeople.dm:
```diff
@@ -116,12 +116,11 @@ Lizard subspecies: ASHWALKERS
 		TRAIT_CAN_STRIP,
 		TRAIT_CHUNKYFINGERS,
-		TRAIT_NOBREATH,
-		TRAIT_VIRUSIMMUNE,
```
as you said, restore this

Mothblocks commented on code/modules/surgery/organs/lungs.dm:
```diff
@@ -562,3 +564,54 @@
+/obj/item/organ/lungs/ashwalker
+	name = "blackened frilled lungs" // blackened from necropolis exposure
+	desc = "Exposure to the necropolis has mutated these lungs to breathe the air of the lava-covered moon below."
+	color = "#964b00" // TODO get proper sprite instead of BROWN
```
is this TODO now

ExcessiveUseOfCobblestone commented on code/modules/surgery/organs/lungs.dm:
```diff
@@ -562,3 +564,54 @@
+	desc = "Exposure to the necropolis has mutated these lungs to breathe the air of the lava-covered moon below."
+	color = "#964b00" // TODO get proper sprite instead of BROWN
+
+/obj/item/organ/lungs/ashwalker/Initialize()
```
initialize has some arguments

coiax commented:
Help the stale bot is bullying me

coiax committed:
Merge branch 'master' into ashwalker-lungs

coiax committed:
Add ashwalker lung icon from CitadelStation

Sprite from https://github.com/Citadel-Station-13/Citadel-Station-13/pull/8159
with sincere thanks to Poojawa for the sprites.

coiax committed:
Readd virus vulnerability, new icon, new tolerance variable

Mothblocks commented:
its still drafted

coiax commented:
@Mothblocks I'm in the testing stage.

coiax committed:
Add toxin support and proper icon file
